src/topology/algebra/mul_action2.lean

```diff
@@ -0,0 +1,162 @@
+
+/-- Class `properly_discontinuous_smul Γ T` says that the scalar multiplication `(•) : Γ → T → T`
+is properly discontinuous, that is, for any pair of compact sets `K, L` in `T`, only finitely many
+`γ:Γ` move `K` to have nontrivial intersection with `L`.
+-/
+class properly_discontinuous_smul (Γ : Type*) (T : Type*) [topological_space T]
+  [has_scalar Γ T] : Prop :=
+(finite_disjoint_inter_image : ∀ {K L : set T}, is_compact K → is_compact L →
+  set.finite {γ : Γ | (((•) γ) '' K) ∩ L ≠ ∅ })
+
+/-- Class `properly_discontinuous_vadd Γ T` says that the additive action `(+ᵥ) : Γ → T → T`
+is properly discontinuous, that is, for any pair of compact sets `K, L` in `T`, only finitely many
+`γ:Γ` move `K` to have nontrivial intersection with `L`.
+-/
+class properly_discontinuous_vadd (Γ : Type*) (T : Type*) [topological_space T]
+  [has_vadd Γ T] : Prop :=
+(finite_disjoint_inter_image : ∀ {K L : set T}, is_compact K → is_compact L →
+  set.finite {γ : Γ | (((+ᵥ) γ) '' K) ∩ L ≠ ∅ })
+
+attribute [to_additive] properly_discontinuous_smul
+
+variables {Γ : Type*} [group Γ] {T : Type*} [topological_space T] [mul_action Γ T]
+
+/-- A finite group action is always properly discontinuous
+-/
+@[priority 100, to_additive] instance fintype.properly_discontinuous_smul [fintype Γ] :
+  properly_discontinuous_smul Γ T :=
+{ finite_disjoint_inter_image := λ _ _ _ _, set.finite.of_fintype _}
+
+export properly_discontinuous_smul (finite_disjoint_inter_image)
+
+export properly_discontinuous_vadd (finite_disjoint_inter_image)
+
+/-- Scalar multiplication by an element of a group `Γ` acting on `T` is a homeomorphism from `T`
+to itself. -/
+def homeomorph.smul {T : Type*} [topological_space T] {Γ : Type*} [group Γ]
+  [mul_action Γ T] [has_continuous_smul₂ Γ T] (γ : Γ) :
+  T ≃ₜ T :=
+{ to_equiv := mul_action.to_perm_hom Γ T γ,
+  continuous_to_fun  := continuous_smul₂ γ,
+  continuous_inv_fun := continuous_smul₂ γ⁻¹ }
+
+/-- Affine-addition of an element of an additive group `Γ` acting on `T` is a homeomorphism
+from `T` to itself. -/
+def homeomorph.vadd {T : Type*} [topological_space T] {Γ : Type*} [add_group Γ]
+  [add_action Γ T] [has_continuous_vadd₂ Γ T] (γ : Γ) :
+  T ≃ₜ T :=
+{ to_equiv := add_action.to_perm_hom T Γ γ,
+  continuous_to_fun  := continuous_vadd₂ γ,
+  continuous_inv_fun := continuous_vadd₂ (-γ) }
+
+attribute [to_additive homeomorph.vadd] homeomorph.smul
+
+/-- The quotient map by a group action is open. -/
+@[to_additive]
+lemma is_open_map_quotient_mk_mul [has_continuous_smul₂ Γ T] :
+  is_open_map (quotient.mk : T → quotient (mul_action.orbit_rel Γ T)) :=
+begin
+  intros U hU,
+  rw [is_open_coinduced, mul_action.quotient_preimage_image_eq_union_mul U],
+  exact is_open_Union (λ γ, (homeomorph.smul γ).is_open_map U hU)
+end
+
+/-- The quotient by a discontinuous group action of a locally compact t2 space is t2. -/
+@[priority 100, to_additive] instance is_t2_of_properly_discontinuous_smul_of_t2 [t2_space T]
```

Comment:
Instance names are not very important, but `t2_space_of_...` would match the statement better.